imports/plugins/core/core/server/Reaction/core.js

```diff
@@ -414,9 +414,19 @@ export default {
    * @method
    * @memberof Core
    * @summary returns the shop which should be used given the current domain
+   * @return {StringId} shopId for the current domain
    */
   getShopIdByDomain() {
     const domain = this.getDomain();
+    const primaryShop = this.getPrimaryShop();
```

Comment:
I wanted to avoid making two queries for this method, but
1. if the condition on line 426 is satisfied, the second query will not execute
2. it is probably ok to assume that `getPrimaryShop` will be called at some other point and could be cached